packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -202,9 +338,18 @@ class GrpcGatewayRouteSpec extends GatewayRouteSpec {
               virtualServiceName: this.routeTarget.virtualServiceName,
             },
           },
+          rewrite: this.match.rewriteRequestHostname === undefined
+            ? undefined
+            : {
+              hostname: {
+                defaultTargetHostname: this.match.rewriteRequestHostname ? 'ENABLED' : 'DISABLED',
+              },
+            },
         },
         match: {
           serviceName: this.match.serviceName,
+          hostname: this.match.hostname?.bind(scope).hostnameMatch,
```

Comment:
Can you actually move `match` above `action`? They're kind of in a weird order right now...